swarmform/core/swarm_dag.py

```diff
@@ -0,0 +1,277 @@
+        """
+
+        self._fw_id = fw_id
+        self._level = level
+        self._fw_info = fw_info
+        self._is_assigned = assigned
+        self._parents = parents
+        self._children = children
+        self._cluster_info = {fw_id: fw_info}
+        self._sequential_ids = []
+        self._parallel_ids = {}
+        self._cluster_space = []
+
+    def set_cluster_space(self, space):
+        self._cluster_space = space
+
+    def get_cluster_space(self):
+        return self._cluster_space
+
+    def set_sequential_ids(self, id_list):
+        self._sequential_ids = id_list
+
+    def get_fw_ids_to_cluster_sequentially(self):
+        return self._sequential_ids
+
+    def set_parallel_ids(self, id_list, key):
+        self._parallel_ids[key] = id_list
+
+    def get_fw_ids_to_cluster_parallely(self):
+        return self._parallel_ids
+
+    def get_fw_id(self):
+        return self._fw_id
+
+    def get_level(self):
+        return self._level
+
+    def get_exec_time(self):
+        if self._fw_info:
+            return self._fw_info['exec_time']
+        else:
+            return None
+
+    def get_num_cores(self):
+        if self._fw_info:
+            return self._fw_info['cores']
+        else:
+            return None
+
+    def get_is_assigned(self):
+        return self._is_assigned
+
+    def get_parents(self):
+        return self._parents
+
+    def get_children(self):
+        return self._children
+
+    def get_cluster_info(self):
+        return self._cluster_info
+
+    def set_cluster_info(self, c_info):
+        """
+        Args:
+            c_info (dict): a dictionary of cluster information containing {fw_id: fw_info}
+        """
+        self._cluster_info = c_info
+
+    def get_fw_ids_to_cluster(self):
+        fw_ids_to_cluster = self._cluster_info.keys()
+        return list(fw_ids_to_cluster)
+
+    def add_parent(self, parent):
+
+        if not isinstance(self._parents, list):
+            self._parents = []
+
+        self._parents.append(parent)
+
+    def add_child(self, child):
+
+        if not isinstance(self._children, list):
+            self._children = []
+
+        self._children.append(child)
+
+    def remove_parent(self, parent_id):
+
+        for parent in self._parents:
+            if parent.get_fw_id() == parent_id:
+                self._parents.remove(parent)
+                break
+
+    def remove_child(self, child_id):
+
+        for child in self._children:
+            if child.get_fw_id() == child_id:
+                self._children.remove(child)
+                break
+
+    def set_fw_info(self, exec_time, cores):
+        self._fw_info = {'exec_time': exec_time, 'cores': cores}
+
+
+class DAG:
+
+    def __init__(self, sf):
+
+        """
+        Args:
+            sf(SwarmFlow): SwarmFlow object
+        """
+
+        self._dag_id = sf.wf_id
+        self._dag_name = sf.name or "Unnamed-WF"
```

Comment:
Shall we rename as Unnamed SF to follow firework's naming format and to maintain a difference in names of a workflow and a swarmflow?

Reply:
noted